### PR TITLE
Updater fixes

### PR DESCRIPTION
**Pull request by gabriel**
Updating working...

You can use the remote to test.. update.json here always points to the most recent build!

```
keybase update --source=remote --url=https://keybase-app.s3.amazonaws.com/update.json
```

### CODE REVIEW

**Comment by gabriel** [comment]
cc @maxtaco 

btw, @MarcoPolo after this goes in you should be able to get update ui notifications (through the delegate ui stuff?)

File: go/install/update.go [hunk 1]
```diff
@@ -234,12 +248,17 @@ func (u *Updater) checkUpdate(sourcePath string, destinationPath string) error {
...
 		}
-		u.G().Log.Info("Moving (existing) %s to %s", dest, tmpDirName)
-		err = os.Rename(dest, tmpDirName)
+		tmpPath := path.Join(os.TempDir(), "KeybaseBackup", tmpFileName)
+		err = libkb.MakeParentDirs(tmpPath)
+		if err != nil {
+			return err
+		}
+		u.G().Log.Info("Moving (existing) %s to %s", dest, tmpPath)
+		err = os.Rename(dest, tmpPath)
```
Comment (maxtaco) [inline go/install/update.go]: do we ever fully remove?

File: go/install/update.go [hunk 2]
```diff
@@ -265,63 +284,68 @@ func (u *Updater) Update(ui libkb.UpdateUI) (update *keybase1.Update, err error)
...
 	if ui == nil {
-		return libkb.NoUIError{Which: "UpdateUI"}
+		err = libkb.NoUIError{Which: "UpdateUI"}
+		return
 	}
 
-	upa := keybase1.UpdatePromptArg{}
-	upa.Update.Version = u.config.Version
-	var upr keybase1.UpdatePromptRes
+	updatePrompt := keybase1.UpdatePromptArg{}
```
Comment (maxtaco) [inline go/install/update.go]: can just:

```go
updatePrompt := keybase1.UpdatePromptArg{ Update : update }
```

**Comment by maxtaco** [comment]
Small comment, looks good :+1: !

File: go/install/update.go (hunk 1 above)
Comment (gabriel) [inline go/install/update.go]: Have an open ticket for this here:
https://keybase.atlassian.net/browse/DESKTOP-165

Putting it in a subdir to make it easier to clear

Comment (maxtaco) [inline go/install/update.go]: Can you put a `TODO` in the comments with a link to the story?

**Pull request merged**